transaction/StronglyConnectedComponents.cpp

```diff
@@ -0,0 +1,125 @@
+/**
+ *   Copyright 2016, Quickstep Research Group, Computer Sciences Department,
+ *     University of Wisconsin—Madison.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#include "transaction/StronglyConnectedComponents.hpp"
+
+#include <vector>
+
+namespace quickstep {
+
+namespace transaction {
+
+StronglyConnectedComponents::StronglyConnectedComponents(const DirectedGraph &directed_graph)
+  : directed_graph_(directed_graph)
+  , is_marked_(directed_graph_.count())
+  , component_ids_(directed_graph_.count())
+  , low_ids_(directed_graph_.count())
+  , preorder_counter_(0)
+  , no_of_components_(0) {
+}
+
+void StronglyConnectedComponents::findStronglyConnectedComponents() {
```

Comment:
As the sole purpose of this class is to find strongly connected components and store them, why not call this function in the constructor? This will ensure that as soon as we create an instance of this class, we will always have all the strongly connected components present.